### PR TITLE
feat: create country field story

### PR DESCRIPTION
**Pull request by jia1**
## Problem

Closes [insert issue #]

## Solution

**Breaking Changes** 

- [ ] Yes - this PR contains breaking changes
    - Details ...
- [ ] No - this PR is backwards compatible  

**Features**:

- Details ...

**Improvements**:

- Details ...

**Bug Fixes**:

- Details ...

## Before & After Screenshots

**BEFORE**:

**AFTER**:

## Deploy Notes

**New environment variables**:

- `env var` : env var details

**New scripts**:

- `script` : script details

**New dependencies**:

- `dependency` : dependency details

**New dev dependencies**:

- `dependency` : dependency details

### CODE REVIEW

**Comment by karrui** [comment]
@jia1 may need you to rename this branch; our CI/CD for react only runs on `form-v2/<branchname` branches

File: shared/constants/countries.ts [hunk 1]
```diff
@@ -0,0 +1,259 @@
+enum Country {
```
Comment (karrui) [inline shared/constants/countries.ts:1]: nit: prefer export as non-default:

` export enum`

File: shared/types/field/base.ts [hunk 2]
```diff
@@ -10,6 +10,7 @@ export enum BasicField {
   ShortText = 'textfield',
   LongText = 'textarea',
   Dropdown = 'dropdown',
+  Country = 'country',
```
Comment (karrui) [inline shared/types/field/base.ts:13]: As discussed, this PR should not have any knowledge of `BasicField.Country`; just create a `CountryField` component that is basically a `BasicField.Dropdown` with hardcoded country field options

File: shared/constants/countries.ts (hunk 1 above)
Comment (jia1) [reply shared/constants/countries.ts:1]: Addressed in https://github.com/opengovsg/FormSG/pull/3705

File: shared/types/field/base.ts (hunk 2 above)
Comment (jia1) [reply shared/types/field/base.ts:13]: Addressed in https://github.com/opengovsg/FormSG/pull/3705

**Pull request closed without merging**